### ISSUE TITLE
Use GitHub Actions and delete Travis / AppVeyor configuration

### ISSUE THREAD

**Issue by tniessen** [nodejs/nan#958]
Some internal discussion suggests that we should migrate all organization repositories to GitHub Actions. As far as I am aware, this is the only remaining repository in the Node.js organization that has a Travis configuration.

This issue is not a duplicate of but likely supercedes https://github.com/nodejs/nan/issues/508.

**Comment by kkoopa** [comment]
Yes, please. Travis has not worked for years and AppVeyor is slow. I presume Windows testing is available as well. I have never used Github Actions.

**Comment by tniessen** [comment]
> I presume Windows testing is available as well. I have never used Github Actions.

Yes. Currently, the list of supported platforms through GitHub-hosted runners includes `ubuntu-22.04`, `ubuntu-20.04`, `windows-2022`, `windows-2019`, `macos-13`, `macos-12`, and `macos-11`. We typically run tests using the aliases `ubuntu-latest`, `windows-latest`, and `macos-latest` so that we don't need to update any configuration as GitHub upgrades their platforms.

**Comment by cclauss** [comment]
Please update the TODO list on this issue.
